### DISCUSSION TITLE
Using Jsdelivr instead of GitHub.

### DISCUSSION THREAD

**Discussion by chobotnice007** [Homebrew/discussions#1117] [Everyday usage]
By default you say on website use this command to install Homebrew: `/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
`. This command will download script from [https://github.com/Homebrew/install/blob/master/install.sh](url) and after that it will run it. I can use command `/bin/bash -c "$(curl -fsSL https://cdn.jsdelivr.net/gh/Homebrew/install@master/install.sh)"
` which will download a script from [https://cdn.jsdelivr.net/gh/Homebrew/install@master/install.sh](url) and after that it will run it. I it is faster to download from Jsdelivr instead of GitHub. 
Why still install Homebrew via script downloaded from GitHub, not from Jsdelivr. Or you have some reservations? Let me know what you think.

**Comment by SMillerDev** [comment]
We don't manage the jsdelivr content while we do control the GitHub repo so we won't be using/recommending this URL.

**Reply by chobotnice007** [reply]
> We don't manage the jsdelivr content while we do control the GitHub repo so we won't be using/recommending this URL.

Jsdelivr is only proxy with can for GitHub and other services.

**Reply by SMillerDev** [reply]
The original page is already world readable with a GitHub CDN backing it. I don't see a reason to use another proxy on top of that, especially one that isn't designed to serve single files but whole JavaScript libraries. It might work, but it adds nothing but complexity.

**Comment by carlocab** [comment]
The install script is tiny (~24K). You can download it from GitHub in less than a tenth of a second. Why would we need to speed up a download that takes a tenth of a second?

**Reply by chobotnice007** [reply]
> The install script is tiny (~24K). You can download it from GitHub in less than a tenth of a second. Why would we need to speed up a download that takes a tenth of a second?

Maybe someone have dial-up.

**Reply by carlocab** [reply]
If you're on dial-up, you're going to have bigger problems than trying to download the install script.

Note that downloading the install script on a dial-up connection should still take about 5 seconds. Is it really that important to optimise a 5 second delay?

**Reply by chobotnice007** [reply]
> If you're on dial-up, you're going to have bigger problems than trying to download the install script.
> 
> Note that downloading the install script on a dial-up connection should still take about 5 seconds. Is it really that important to optimise a 5 second delay?

Yes, it is.

**Reply by tlk** [reply]
> Maybe someone have dial-up.

When the install script has been downloaded it will go on and clone a git repository that is about 15.000 (fifteen thousand) times larger than the script.

It does make sense to want to reduce the bandwidth usage, but the download speed of the install script is insignificant.

**Reply by chobotnice007** [reply]
> > Maybe someone have dial-up.
> 
> When the install script has been downloaded it will go on and clone a git repository that is about 15.000 (fifteen thousand) times larger than the script.
> 
> It does make sense to want to reduce the bandwidth usage, but the download speed of the install script is insignificant.

Yes, that is bigger problem.
